packages/clay-core/src/picker/Option.tsx

```diff
@@ -92,9 +101,14 @@ export function Option({
 
 	const isFocus = isFocusVisible();
 
+	const filteredOtherProps = Object.fromEntries(
+		Object.entries(otherProps).filter(([key]) => key.startsWith('data-'))
+	) as Record<DataAttributeKey, any>;
```

Comment:
@cgmonte I'm thinking about this a bit, I think we don't need a runtime check, in theory since we adopted TypeScript we let TypeScript handle the validation mainly if you pass an undefined value the build will fail.

Another thing is that we can probably add more new APIs soon so I think it's safe here not to need this validation for now.

Reply:
The thing is that TS is not strict with attributes that aren't valid JavaScript identifiers, but are valid in JSX. So it does not quite check attribute names like `data-*`, `aria-*` or even `anything-*`, so `data-${string}: any` isn't doing anything for us. For example, after declaring like this:
```
type Props = {
	// ...
	[key: `data-${string}`]: any;
};
```
I can still pass something like this without getting TS errors:
```
	<Picker>
		<Option anything-attribute="any value" />
	</Picker>
```
The above information can be found in [TypeScript documentation](https://www.typescriptlang.org/docs/handbook/jsx.html#attribute-type-checking), where it reads like:

> Note: If an attribute name is not a valid JS identifier (like a data-* attribute), it is not considered to be an error if it is not found in the element attributes type.

I have tried other strategies to make this work with TypeScript, but could not get it right. There is probably a way, using helper types or explicitly declaring `ElementAttributesProperty`, but I am not sure if this is the way to go. 

Let me know if validating this is really necessary. I could not find other examples of such validations, using string templates, in Clay's codebase.

Reply:
Interesting, I didn't know about this problem... so this really won't help at all, not even for the description of the type for the developer, unfortunately.